ggshield/core/check_updates.py

```diff
@@ -63,13 +63,37 @@ def check_for_updates() -> Optional[str]:
     try:
         resp = requests.get(
             "https://api.github.com/repos/GitGuardian/GGShield/releases/latest",
+            headers={
+                "Accept": "application/vnd.github+json",
+                "User-Agent": f"GGShield {__version__}",
+                "X-Github-Api-Version": "2022-11-28",
+            },
             timeout=CHECK_TIMEOUT,
         )
     except Exception as e:
         logger.error("Failed to connect to api.github.com: %s", e)
         return None
 
     if resp.status_code != 200:
+        # Handle GitHub rate limit responses gracefully
+        # https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting
+        if int(resp.headers.get("X-RateLimit-Remaining", -1)) == 0:
+            logger.debug("GitHub rate limit exceeded - rescheduling update check")
+
+            # Reset the next update check based on when the GH API quota resets
+            check_at = int(resp.headers.get("X-RateLimit-Reset", -1.0)) - 24 * 60 * 60
```

Comment:
nitpick: since we convert to int, we can make the default value an int too.
```suggestion
            check_at = int(resp.headers.get("X-RateLimit-Reset", -1)) - 24 * 60 * 60
```